src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValue.cs

```diff
@@ -181,13 +181,15 @@ internal static JsonValue CreateFromTypeInfo<T>(T value, JsonTypeInfo<T> jsonTyp
 
         internal static JsonValue? CreateFromElement(ref readonly JsonElement element, JsonNodeOptions? options = null)
         {
-            if (element.ValueKind is JsonValueKind.Null)
+            JsonValueKind kind = element.ValueKind;
```

Comment:
Nit: could this be refactored to switch statement?

Reply:
Seems a bit more complicated to understand because of the method which `[DoesNotReturn]`

```csharp
        internal static JsonValue? CreateFromElement(ref readonly JsonElement element, JsonNodeOptions? options = null)
        {
            switch (element.ValueKind)
            {
                case JsonValueKind.Null:
                    return null;

                case JsonValueKind.Object:
                case JsonValueKind.Array:
                    // Force usage of JsonArray and JsonObject instead of supporting those in an JsonValue.
                    ThrowHelper.ThrowInvalidOperationException_NodeElementCannotBeObjectOrArray();

                    // The above method will throw, but the compiler doesn't know that.
                    return null;

                default:
                    return new JsonValueOfElement(element, options);
            }
        }
```

still want to go ahead?

Reply:
Yep, this is fine.